File: lesson01/app/src/main/java/br/odb/nehe/lesson01/GL2JNIActivity.java
package br.odb.nehe.lesson01;

import android.app.Activity;
import android.content.res.AssetManager;
import android.os.Bundle;


public class GL2JNIActivity extends Activity {

    GL2JNIView mView;
    static AssetManager assets;
    @Override protected void onCreate(Bundle icicle) {
        super.onCreate(icicle);

        assets = getAssets();
        GL2JNILib.onCreate( assets );
        mView = new GL2JNIView(getApplication());
        setContentView(mView);
    }

    @Override protected void onPause() {
        super.onPause();
        mView.onPause();
    }

    @Override protected void onResume() {
        super.onResume();
        mView.onResume();
    }

    @Override
    protected void onDestroy() {

        mView = null;
        super.onDestroy();
    }
}

File: lesson02/app/src/main/java/br/odb/nehe/lesson02/GL2JNIActivity.kt
package br.odb.nehe.lesson02

import android.app.Activity
import android.os.Bundle

class GL2JNIActivity : Activity() {
	
	private var mView: GL2JNIView? = null

	override fun onCreate(icicle: Bundle?) {
		super.onCreate(icicle)
		GL2JNILib.onCreate(assets)
		mView = GL2JNIView(application)
		setContentView(mView)
	}

	override fun onPause() {
		super.onPause()
		mView!!.onPause()
	}

	override fun onResume() {
		super.onResume()
		mView!!.onResume()
	}

	override fun onDestroy() {
		GL2JNILib.onDestroy()
		super.onDestroy()
	}
}